### PR TITLE
fix(desktop): 「已工作」时长丢失首段思考时间,段起点改锚上一边界

### PR DESCRIPTION
**Pull request by 0xmariowu**
### 摘要

「已工作 Xs」低报:段时长起点锚在段内第一个动作自带的时间戳上,而一次性到达(非流式)的 thinking 块 `createdAt`≈结束时刻,模型真正思考的几秒被整段丢掉。

真实会话数据复现(桌面端本地库原始时间戳):用户消息 → **6.06s** 后 thinking 落库(自记 `durationMs` 仅 109ms)→ **2ms** 后正文落库。首段实际耗时 ~6s,界面显示「已工作 1s」。同一轮里外层总表与内层段相加也对不上(外层 29s,内层 1s + 26s = 27s),缺口就是被丢掉的思考时间。

修复:段起点改为优先锚**上一个边界**(用户消息 / 上一句正文)——与「正在工作…」活表已经在用的墙钟口径一致;边界缺失(渲染窗口截断)或时序异常(rewind 改序)时退回原有段内锚点,行为不劣于修复前。

### 变更类型

- [ ] `feat` 新功能
- [x] `fix` 缺陷修复
- [ ] `refactor` / `perf` 重构或性能优化
- [ ] `docs` / `test` / `chore` 文档、测试或工程维护
- [ ] 其他:

### 范围

- 关联 Issue / 需求:无(可按需补开)
- 本 PR 包含:`MessageStream.tsx` 工作组分组的段起点锚定逻辑(`createWorkGroup` 及四个调用路径透传上一边界时间戳);新增回归测试 `workGroupDurationAnchor.test.ts`;更新 `buildRenderItemsKeyStability.test.ts` 中两处按旧口径断言的时长预期。
- 明确不包含:thinking 块 `durationMs` 在 translator 层的计量(那是数据源问题,本 PR 只修展示层口径);「已工作」文案与视觉样式。
- 用户可见变化:「已工作 Xs」与「正在工作…」计时数值更准确——纯思考段不再显示 1s,内层各段相加与外层总表对齐。
- 是否存在 breaking change:无

### UI 变化

不涉及:命中 UI 代码路径,但无视觉、交互、布局或文案变化,仅时长数值的计算语义修正(显示格式、组件结构均不变)。

- 引用的设计规范:不涉及(同上理由)

### 自动验证

```text
pnpm --filter desktop run --if-present typecheck
结果:通过(无输出)

npx vitest run src/renderer/__tests__/workGroupDurationAnchor.test.ts \
  src/renderer/__tests__/buildRenderItemsKeyStability.test.ts \
  src/renderer/__tests__/workGroupRunningSubagent.test.ts \
  src/renderer/__tests__/workGroupBlockLivePreview.test.tsx
结果:Test Files 4 passed (4) / Tests 88 passed (88)

pnpm test:unit
结果:exit 0 全绿(48 workspace PASS)

pnpm check:dco
结果:DCO check passed: 1 commit signed off
```

新增测试先在未修复代码上运行确认为红(首段算出 111ms 而非 6062ms),修复后转绿。

说明:本地全量 `pnpm test:unit` 需以 `GIT_CONFIG_GLOBAL=/dev/null` 运行——开发机全局 gitignore 含 `.env`,会让 `gitSnapshotService.test.ts` 在临时仓库 `git add .env` 时失败;该干扰与本 PR 无关(修复前的 main 同样失败),中和全局配置后该文件 17/17 通过。

### 手工验证

不涉及实机 UI 操作:修复对象是纯函数分组逻辑,新增测试的时间线夹具直接镜像真实会话数据库的原始时间戳(user → +6.06s thinking(自记 109ms) → +2ms 正文)。

### 未执行的验证

实机目检渲染效果未执行:改动不触及组件渲染结构与样式,时长数值路径已由 88 个相关单测覆盖(含本 PR 新增 4 个)。

### 风险分类

- [x] 无已知风险
- [ ] SQLite / migration
- [ ] system prompt
- [ ] 协议兼容
- [ ] 权限 / 安全 / 用户数据
- [ ] 原生层 / fingerprint / OTA
- [ ] 跨平台差异
- [ ] 其他:

### 影响与回滚

- 影响范围:仅聊天时间线工作组的「已工作 / 正在工作」时长数值;不落库、不走 IPC、不影响协议与持久数据。
- 回滚 / 降级方式:revert 单 commit 即回到原行为。

### 提交前检查

- [x] 已 review 完整 diff
- [x] 每个 commit 都带 DCO 签名(`git commit -s`,见 [DCO](../DCO))
- [x] UI 改动已在「UI 变化」注明引用的设计规范章节(不涉及 UI 则跳过)
- [x] 未提交凭证、令牌或授权文件
- [x] 已补充必要文档(测试文件头注释记录了复现与口径说明)
- [x] 已确认测试结果或说明未执行原因

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## 界面效果(修复前 → 修复后)

组件结构与样式零变化,仅「已工作 Xs」数值语义修正。以真实复现会话的时间戳为例
(用户消息 → +3s 纯 thinking → 正文 → +26s 工具段 → 最终正文):

```html
<div style="font: 13px/1.7 -apple-system, 'PingFang SC', sans-serif; color:#333; max-width:560px">
  <p style="color:#999; margin:0 0 4px">修复前 — 首段思考被丢,内层相加 ≠ 外层</p>
  <div style="border:1px solid #e2e2e2; border-radius:10px; padding:10px 14px; margin-bottom:14px">
    <div>≋ 已工作 29s ▾</div>
    <div style="margin-left:16px; color:#c0392b">≋ 已工作 1s ▸ ← 实际思考 ~3s,只记了到达窗口</div>
    <div style="margin-left:16px">我先快速摸一下代码规模和质量证据,再给评价。</div>
    <div style="margin-left:16px">≋ 已工作 26s ▸</div>
    <div style="color:#c0392b; margin-top:2px">1s + 26s = 27s ≠ 29s(缺口 = 被丢的思考时间)</div>
  </div>
  <p style="color:#999; margin:0 0 4px">修复后 — 段起点锚上一边界,账目对齐</p>
  <div style="border:1px solid #e2e2e2; border-radius:10px; padding:10px 14px">
    <div>≋ 已工作 29s ▾</div>
    <div style="margin-left:16px; color:#27ae60">≋ 已工作 3s ▸ ← 从用户消息起表</div>
    <div style="margin-left:16px">我先快速摸一下代码规模和质量证据,再给评价。</div>
    <div style="margin-left:16px">≋ 已工作 26s ▸</div>
    <div style="color:#27ae60; margin-top:2px">3s + 26s = 29s = 外层总表 ✓</div>
  </div>
</div>
```

「正在工作…」活表(流式态)同口径:从上一边界起跳,与完成态数字连续,无视觉变化。

### CODE REVIEW

**Comment by MagicLizi** [comment]
@0xmariowu 👋 这个 PR 还有 1 条 review conversation 没 resolve(apps/desktop/src/renderer/components/chat/MessageStream.tsx),auto-review 因此暂时跳过、没法继续审查 / 合并。

如果你已经按评论改完或回应了,请到对应 thread 上点 **Resolve conversation**;全部 resolve 后,下一轮 auto-review 会自动重新审查这个 PR。

File: apps/desktop/src/renderer/components/chat/MessageStream.tsx [hunk 1]
```diff
@@ -1375,9 +1381,18 @@ function createWorkGroup(
...
+  prevBoundaryTs: number | null = null,
 ): Extract<RenderItem, { type: 'work_group' }> {
   const firstActivity = run.find((it) => it.type !== 'message' || it.message.role === 'thinking');
-  const startTs = workRunStartTs(firstActivity ?? run[0]);
+  const anchorTs = workRunStartTs(firstActivity ?? run[0]);
+  // 段起点优先锚上一个边界(用户消息 / 上一句正文),与「正在工作…」活表的墙钟
+  // 口径一致:一次性到达的 thinking 块 createdAt≈结束时刻,只用段内锚点会把
+  // 模型思考整段丢掉(实际 6s 显示 1s,内层相加也对不上外层总表)。边界缺失
+  // (窗口截断)或时序异常(rewind 改序)时退回段内锚点。
+  const startTs =
```
[the remark this answers is not in the document]
Comment (0xmariowu) [reply apps/desktop/src/renderer/components/chat/MessageStream.tsx:1394]: 有道理,两处 `startedAtMs` 的注释(MessageStream 的 item 字段 + WorkGroupBlock 的 props)已在 5af8badc 更新,点明起点优先是上一边界、可能早于首个活动,与 durationMs 同源。

**Comment by MagicLizi** [comment]
命中 UI 路径（`apps/desktop/src/renderer/__tests__/buildRenderItemsKeyStability.test.ts`、`apps/desktop/src/renderer/__tests__/workGroupDurationAnchor.test.ts`、`apps/desktop/src/renderer/components/chat/MessageStream.tsx` 等）但 description 未附界面效果证据——建议补充改动后效果：截图/录屏，或改动后界面的 HTML 页面（```html 代码块、.html 附件或在线预览链接），便于确认界面符合 DESIGN.md 设计规范。

**Comment by dashhuang** [comment]
这个观察很细致。能从一次性 thinking 的时间戳语义里定位到首段思考时间被整段漏掉，不只是修正一个显示数字，也让「已工作 / 正在工作」更贴近用户真实等待时间、内外层计时口径更一致。这个修复意义很大，感谢把复现、完成态 / 流式态和回退场景都补得这么扎实。

**Comment by 0xmariowu** [comment]
@MagicLizi 两件都已处理:
1. MessageStream.tsx 那条 review conversation 已按建议在 5af8badc 更新注释并 resolve;
2. PR 描述已补「界面效果」小节,用真实复现会话的数字做了修复前/后 HTML 对比(组件结构与样式零变化,仅数值语义)。

@dashhuang 谢谢肯定。这个问题是从桌面端本地库的原始时间戳里对出来的(thinking 落库时刻与正文只差 2ms,但自记时长仅 109ms),数据都在 PR 描述里。后面想继续给 Cindy 提一些体验相关的改进,有方向问题会先开 issue 讨论。

**Review by MagicLizi — APPROVED** [review]
Pass: timing logic fix correctly anchors segment start from prev boundary (user message / preceding text). All edge cases handled (missing boundary, time anomaly, multi-segment, streaming). 70+18 tests pass. No security findings. DESIGN.md N/A (no visual change, only numeric semantics).
